src/PostgREST/Request/DbRequestBuilder.hs

```diff
@@ -152,21 +153,25 @@ findRel schema allRels origin target hint =
   case rel of
     []  -> Left $ NoRelBetween origin target
     [r] -> Right r
-    rs  ->
-      -- Return error if more than one relationship is found, unless we're in a
-      -- self reference case.
-      --
-      -- Here we handle a self reference relationship to not cause a breaking
-      -- change: In a self reference we get two relationships with the same
-      -- foreign key and relTable/relFtable but with different
-      -- cardinalities(m2o/o2m) We output the O2M rel, the M2O rel can be
-      -- obtained by using the origin column as an embed hint.
-      let [rel0, rel1] = take 2 rs in
-      if length rs == 2 && relLink rel0 == relLink rel1 && relTable rel0 == relTable rel1 && relFTable rel0 == relFTable rel1
-        then note (NoRelBetween origin target) (find (\r -> relType r == O2M) rs)
-        else Left $ AmbiguousRelBetween origin target rs
```

Comment:
Could you explain that change a little bit?

Where is the `note (NoRelBetween...)` gone? Was that dead code?

Reply:
This is the code related to: https://github.com/PostgREST/postgrest/issues/1643#issuecomment-721417728. It will be deleted when I fix that issue.

> note (NoRelBetween...) gone?

Yes, the `note` was just to handle the case were the `find  relType r == O2M` returned Nothing(which couldn't happen). Now that there's a pattern match for 2 exact elements in the list that's not needed anymore.

Reply:
> This is the code related to: [#1643 (comment)](https://github.com/PostgREST/postgrest/issues/1643#issuecomment-721417728). It will be deleted when I fix that issue.

Ok, I had a feeling that this piece of code is something that will possibly change once we fix some things.

My main concern was, that it still is only refactor at this point and no implied feature or fix. Just to double check - is that correct?


> Yes, the `note` was just to handle the case were the `find relType r == O2M` returned Nothing(which couldn't happen). Now that there's a pattern match for 2 exact elements in the list that's not needed anymore.

I think those impossible branches / pattern matches are kind of a result of wrong typing. Wrong in the sense of "typing would allow cases, that just don't happen in reality". Now that we fixed it for these types - I'm really curious how the codecov report comes back. There should really not be any impossible branches showing up.

Reply:
> My main concern was, that it still is only refactor at this point and no implied feature or fix. Just to double check - is that correct?

Yes, I've made sure the logic is the same, no break(there are tests as well).

> Now that we fixed it for these types - I'm really curious how the codecov report comes back. 

You know, by mistake I removed the secrets of my own CircleCI setup yesterday. I've now added the codecov token back, hope the report shows now.